cmd/soroban-cli/src/config/locator.rs

```diff
@@ -468,14 +468,22 @@ impl KeyType {
 }
 
 pub fn global_config_path() -> Result<PathBuf, Error> {
```

Comment:
I think this should return `[Option<PathBuf>; 2]`. Same with the local config. We already have the notion of prioritizing local over global. And so then we can add additional dimension to each. You would need to update `get_local_and_global` but the other functions expect locations in priority order so everything should work.

Reply:
All places that actually need both global and local config are calling `local_and_global`, which is returning `Result<[Location; 2], Error>`, so they can display all configs. I fail to see why changing the return type like you suggested is something we need to do.

Reply:
The point is that you want to collect all keys from all configs. For example you have a key for Alice in .stellar, and Bob in .soroban. Then others in the other two global configs. 

So you need to look for the keys that prioritize local then global as before then also stellar then soroban for each.

Reply:
I don't think that's true. Right now, we prioritize .stellar over .soroban, and show an warning in case both exists at the same time: https://github.com/stellar/stellar-cli/blob/main/cmd/soroban-cli/src/utils.rs#L87-L89

I followed the same idea here, so it works similarly for both local and global configs. https://github.com/stellar/stellar-cli/pull/1681/files#diff-09da31e3b060c449882b17935698b71f2d9a1e1cd5a8198c135b594da32a4788R484-R486

Reply:
I am not at my computer but if Bob didn't exist in stellar but did in sororban this would find it? When I looked before it seemed to pick stellar and ignore soroban

Reply:
I am not at my computer but if Bob didn't exist in stellar but did in sororban this would find it? When I looked before it seemed to pick stellar and ignore soroban

Reply:
> I am not at my computer but if Bob didn't exist in stellar but did in sororban this would find it?

No, it wouldn't, but this is already the case. You would get a warning like `tracing::warn!("the .stellar and .soroban config directories exist at path {pwd:?}, using the .stellar");` though.

Reply:
We're always favoring `.stellar`, and not fetching both directories. If we decide we should, then that's a change for another pr, because it affects the current behavior.

Reply:
Then I think we should add a migration option. As @leighmcculloch said users could already have keys in their old soroban configs and currently not being able to resolve them is a bug IMO. And this seems like the perfect PR to fix that or add a migration option because we don't want users to ever have to manually inspect these directories.

Reply:
We discussed this on slack and decided that it's not worth supporting the edge case where both directories can exist; the warning should be enough to give users a hint on what's being used.